### PR TITLE
fix for #152

### PR DESCRIPTION
**Pull request by zcaudate**
Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- TESTED You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [x] You've updated the [changelog](../CHANGELOG.md)(that only applies to user-visible changes)

Thanks!

*If you're just starting out to hack on nREPL you might find this [section of its
manual][1] extremely useful.*

[1]: https:/nrepl.org/nrepl/hacking_on_nrepl.html

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### New features
 
+* [#152](https://github.com/nrepl/nrepl/issues/60): Bug fix for 'close' operation to kill session thread
```
Comment (bbatsov) [inline CHANGELOG.md]: Seems you forgot the update that link to point to the actual issue. Bug fixes should go under "Bugs fixed", not under new features.

**Comment by bbatsov** [comment]
The suggested fix seems reasonable to me, but I'd like to hear what the other members of @nrepl/nrepl think about it. 

It might be a good idea to add a few inline comments about the session threads lifecycle.

**Comment by zcaudate** [comment]
I've added in a comment regarding `session-exec` and fixed the CHANGELOG

File: src/clojure/nrepl/middleware/session.clj [hunk 2]
```diff
@@ -165,26 +165,32 @@
...
+    ;; This map is added to the meta of the session object by `register-session`,
+    ;; it contains functions that are accessed by `interrupt-session` and `close-session`.
+    {:interrupt (fn [exec-id]
+                  ;; nil means interrupt whatever is running
+                  ;; returns :idle, interrupted id or nil
                   (let [current @running]
                     (cond
                       (nil? current) :idle
-                      (and (or (nil? exec-id) (= current exec-id)) ; cas only checks identity, so check equality first
+                      (and (or (nil? exec-id) (= current exec-id)) ;; cas only checks identity, so check equality first
```
Comment (bbatsov) [inline src/clojure/nrepl/middleware/session.clj]: This should be just `;`, as it's an inline comment.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@
 
 * [#60](https://github.com/nrepl/nrepl/issues/60): Implemented EDN transport.
 
+### Bugs fixed
+
+* [#152](https://github.com/nrepl/nrepl/issues/152): Bug fix for 'close' operation to kill session thread
```
Comment (bbatsov) [inline CHANGELOG.md]: I guess a better wording would be "Kill session threads when closing sessions." (or something like this).

**Comment by bbatsov** [comment]
The patch looks good to me. @cgrand @cichli @SevereOverfl0w Any thoughts from you?

**Comment by SevereOverfl0w** [comment]
Just had a quick look at the code, and calling `(close)` seems like the right approach here based on my understanding that each session has a persisting thread that backs it.

**Comment by zcaudate** [comment]
what's the progress for the PR?

**Comment by bbatsov** [comment]
@zcaudate I had left a couple of tiny remarks and I was just waiting for them to be addressed. The PR is pretty much good to go.

**Comment by bbatsov** [comment]
@zcaudate ping :-)

**Comment by zcaudate** [comment]
@bbatsov. I didn’t realise you were waiting on my end. Will fix the changes asap.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**